libraries/aws_ssm_resource_compliance_summary.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsSsmResourceComplianceSummary < AwsResourceBase
+  name 'aws_ssm_resource_compliance_summary'
+  desc 'Verifies settings for a SSM Resource Compliance Summary'
+
+  example "
+    describe aws_ssm_resource_compliance_summary(resource_id: 'resource-id') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:compliance_type,       field: :compliance_type)
+             .register_column(:resource_type,         field: :resource_type)
+             .register_column(:resource_id,           field: :resource_id)
+             .register_column(:status,                field: :status)
+             .register_column(:overall_severity,      field: :overall_severity)
+             .register_column(:execution_summary,     field: :execution_summary)
+             .register_column(:compliant_summary,     field: :compliant_summary)
+             .register_column(:non_compliant_summary, field: :non_compliant_summary)
+             .register_column(:compliance_item,       field: :compliance_item)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
```

Comment:
This resource seems to use too much computing for the sake of importing the compliant and non-compliant severity count from the `@aws.ssm_client.list_resource_compliance_summaries` method. And the result will be a mix of 2 method responses.
I suggest using the `@aws.ssm_client.list_compliance_items` method only and counting the severities with static methods. This will prevent pulling all compliance summaries every time testing a single resource.
The name of the resource can be changed to `aws_ssm_compliance_items` to differentiate it better from the `aws_ssm_resource_compliance_summaries`.
In case I am missing some background discussions:
@skpaterson, @rmoles